nni/compression/experiment/config/utils.py

```diff
@@ -0,0 +1,152 @@
+    if not target or (isinstance(target, (int, float)) and target == 0):
+        return (0., 0.)
+    elif isinstance(target, float):
+        assert 0. < target < 1.
+        return (-0.1 - target, -50.)
+    elif isinstance(target, int):
+        assert 0 < target < origin
+        return (-0.1 - target / origin, -50.)
+    else:
+        raise NotImplementedError('Currently only supports setting the lower limit.')
+
+# hard code and magic number for metric reward function
+def _metric_theta_helper(target: Optional[float], origin: float) -> Tuple[float, float]:
+    if not target:
+        return (-0.85, 50.)
+    elif isinstance(target, float):
+        assert 0. <= target <= 1.
+        return (0.1 - target, 50.)
+    else:
+        raise NotImplementedError('Currently only supports setting the lower limit.')
+
+def _summery_module_names(model: Module,
+                          module_types: List[Union[Type[Module], str]],
+                          module_names: List[str],
+                          exclude_module_names: List[str]) -> List[str]:
+    _module_types = set()
+    _all_module_names = set()
+    module_names_summery = set()
+    if module_types:
+        for module_type in module_types:
+            if isinstance(module_type, Module):
+                module_type = module_type.__name__
+            assert isinstance(module_type, str)
+            _module_types.add(module_type)
+
+    # unfold module types as module names, add them to summery
+    for module_name, module in model.named_modules():
+        module_type = type(module).__name__
+        if module_type in _module_types:
+            module_names_summery.add(module_name)
+        _all_module_names.add(module_name)
+
+    # add module names to summery
+    if module_names:
+        for module_name in module_names:
+            if module_name not in _all_module_names:
+                # need warning, module_name not exist
+                continue
+            else:
+                module_names_summery.add(module_name)
+
+    # remove module names in exclude_module_names from module_names_summery
+    if exclude_module_names:
+        for module_name in exclude_module_names:
+            if module_name not in _all_module_names:
+                # need warning, module_name not exist
+                continue
+            if module_name in module_names_summery:
+                module_names_summery.remove(module_name)
+
+    return list(module_names_summery)
+
+def cc_cv2ss(config: CompressionConfig, vessel: CompressionVessel):
+    search_space = {}
+    model, _, evaluator, dummy_input, _, _, _, _ = vessel.export()
+    flops, params, results = count_flops_params(model, dummy_input, verbose=False, mode='full')
+    metric = evaluator(model)
+
+    module_names_summery = _summery_module_names(model, config.module_types, config.module_names, config.exclude_module_names)
+    for module_name in module_names_summery:
```

Comment:
My understanding of how this works: user selects a pruner, we help them generate a search space, which contains hyper-parameters of this pruner. We tune the hyper-parameters until satisfying their constraints.

My question is: how do we know what's the appropriate search space? For example, why can't we select the pruner automatically?